### ISSUE TITLE
Container image for 0.11.0

### ISSUE THREAD

**Issue by fntlnz** [iovisor/bpftrace#1522]
### What reproduces the bug?
Was looking into updating kubectl-trace to the latest bpftrace 0.11.0 however I can't find the `quay.io/iovisor/bpftrace:v0.11.0` image that we would need to use as base image.

I see that there's automation setup here to publish the images and in fact we do have the images for `v0.9.4` and `v.0.10.0` for example however the `v0.11.0` one is missing.

Anyone has idea why? I'm here to help if there's anything I can do to get the image published.

**Comment by fbs** [comment]
It looks like the 0.11 build never pushed to quay:

https://github.com/iovisor/bpftrace/runs/874428049?check_suite_focus=true

But master did:

https://github.com/iovisor/bpftrace/runs/874400231?check_suite_focus=true

![Screenshot 2020-09-19 at 20 01 49](https://user-images.githubusercontent.com/474727/93685905-3c356680-faa2-11ea-930d-5475ca805123.png)

I think this is the issue:

```
    - name: Package docker image and push to quay.io
      if: >
        matrix.env['TYPE'] == 'Release' &&
        github.ref == 'refs/heads/master' &&
        github.repository == 'iovisor/bpftrace'
      run: >
        ./docker/scripts/push.sh
```

For a tag the ref is `refs/tags/v0.11` I think.

Looks like 8a2892083 is the culprit,  which landed a few days after the 0.10 release. That commit was added to avoid having forks try to upload to quay.

Not sure if we need the `github.ref` check, just checking for `github.repository` should be enough I think.

To fix this I think we have to land a fix in master. Then backport that into a v0.11 branch and do a v0.11.1 release.

Unless we can manually tag an image in quay. I'd have to check whether that is possible (and I'm not sure I have access)

cc @mmisono @dalehamel

**Comment by fbs** [comment]
Based on https://docs.quay.io/guides/tag-operations.html it seems possible to changes tags. So we'd just have to tag the master build. But I don't have access

**Comment by fntlnz** [comment]
I do have the necessary permissions on quay, if you tell me the tags we need to make I can make this happen. I could do the tag understanding myself but you are the one cutting the releases so your eyes are ways more reliable than mines here :) Either that or I can give you access to quay.

**Comment by fbs** [comment]
If you can give me access that would be great! :)

**Comment by dalehamel** [comment]
> For a tag the ref is refs/tags/v0.11 I think.

For tagged refs, it should actually push 2 builds. One for the branch (eg, maste), and one for the tag. So i'm not sure why it hasn't pushed it, it might be how the repo or github workflow is configured.

You're correct we can just manually tag the correct SHA here, but I think we should also figure out why it isn't publishing on tagged builds. Unfortunately, this is difficult for me to test as I'm not an owner on the bpftrace repo (though I recall testing it on my own fork...)

@fbs i have invited you to the iovisor org on quay.io as an owner, you should be able to push tags once you accept.

I can lend a hand in fixing this for future tagged revisions if required.

**Comment by dalehamel** [comment]
https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#onpushpull_requestbranchestags

It *should* be building all tags as far as I can tell...

**Comment by fbs** [comment]
I'm pretty sure the issue is the filter that got added:

```
    - name: Package docker image and push to quay.io
      if: >
        matrix.env['TYPE'] == 'Release' &&
        github.ref == 'refs/heads/master' &&
        github.repository == 'iovisor/bpftrace'
      run: >
        ./docker/scripts/push.sh
```

Which limits it to the master branch in iovisor/bpftrace. If we update it to `master branch` or `tags` it should work.

**Comment by fbs** [comment]
I've just added a `v0.11.0` tag on quay based on the `a9ba414ea8212e825cd48ac536aba66af76c0cfc` image on quay. It seems to be ok and run without issues on fed31.

I'm tempted to backport e52519b72c43b9424c06dd4a37d7a10e7386bf54 into a 0.11.1 release, as not being able to use END probes is a bit lame.

**Comment by dalehamel** [comment]
> I'm tempted to backport e52519b into a 0.11.1 release, as not being able to use END probes is a bit lame.

Good idea

**Comment by dalehamel** [comment]
> Which limits it to the master branch in iovisor/bpftrace. If we update it to master branch or tags it should work.

Ah yes you're correct @fbs, I hadn't seen https://github.com/iovisor/bpftrace/commit/d308fe55ab10bb648cada3fe0e345dad4294049(cc @danobi ), I didn't realize the branch filtering had changed, we need to allow it to accept all tags. I believe a github action build for the tag is just skipping this step as you indicate.

It would be nice to fix this issue completely, so maybe we need to have a separate PR to fix the filter to allow all tags, eg, allow "refs/tags/*" OR refs/heads/master, just need to change the filter syntax to accept this, I think there is a way to do `||` in the expression, and probably wildcard it as well.

**Comment by dalehamel** [comment]
oh wait ,that PR just changes the line endings...

**Comment by dalehamel** [comment]
I think we want something like:

```
        matrix.env['TYPE'] == 'Release' &&
        (github.ref == 'refs/heads/master' || startsWith(github.ref, 'refs/tags')) &&
        github.repository == 'iovisor/bpftrace'
```

But need to verify the syntax with github actions.

**Comment by dalehamel** [comment]
Trying this out on https://github.com/iovisor/bpftrace/pull/1525, i'll try and push it for my own repo.

**Comment by dalehamel** [comment]
Also wondering why #1292 was added, maybe we should just revert it? The PR description doesn't state the reason why only master should be allowed to push, since the ref we push to is parsed out of the ref https://github.com/iovisor/bpftrace/blob/master/docker/scripts/push.sh#L29

#1292 there is inline documentation of how push functions:

https://github.com/iovisor/bpftrace/blob/master/docker/scripts/push.sh#L14-L16

This is fine, because no one can update master other than repo admins who can merge to master, so if #1292 was done to protect the master branch, it doesn't do that - this is handled through github permission structure already, as only builds on the master branch can update the master / latest ref, and in fact, this is also right in the script:

https://github.com/iovisor/bpftrace/blob/master/docker/scripts/push.sh#L53

Since pull requests run against a user's own fork of the repo, and not iovisor's fork, they can only push to their *own* master branch for quay, since their builds don't even run in the context of the iovisor/bpftrace repo, only people with push access to the repo can do that.

**Comment by dalehamel** [comment]
@mmisono what do you think?^

**Comment by fbs** [comment]
As the discussion is happing in a few issues now.

The changes was added to:

- prevent actions on forks from failing as they cannot push the image
- prevent non-master branches from being released as builds (like the uwrite branch).

Once #1525 lands I'll create a 0.11_release branch of the v0.11.0 tag and backport 1525 and e52519b72c43b9424c06dd4a37d7a10e7386bf54 so we have a proper release in quay

**Comment by fbs** [comment]
v0.11.1 has just been tagged and images are starting to show up:

![Screenshot 2020-09-22 at 21 09 31](https://user-images.githubusercontent.com/474727/93926629-47d59700-fd07-11ea-9b85-8eef3e45dc8a.png)

**Comment by fbs** [comment]
```
sudo docker run -v $PWD/out:/output quay.io/iovisor/bpftrace:v0.11.1 bpftrace -V
bpftrace v0.11.1
```
